lib/wallet/api/http/Cardano/Wallet/Api/Types/Primitive.hs

```diff
@@ -275,3 +284,20 @@ instance FromJSON (ApiT TxMetadata) where
 
 instance ToJSON (ApiT TxMetadata) where
     toJSON = metadataToJson TxMetadataJsonDetailedSchema . getApiT
+
+instance FromJSON (ApiT (Hash "ScriptIntegrity")) where
+    parseJSON value = do
+        (hrp, bytes) <- parseJSON value >>=
+            parseBech32 "Malformed policy key"
+        when
+            (hrp /= [humanReadablePart|script_data|])
+            $ fail "expected a bech32 script_data hash"
+        when
+            (BS.length bytes /= 32)
+            $ fail "expected a bech32 script_data hash of 32 bytes"
+        pure $ ApiT $ Hash bytes
```

Comment:
ℹ️ This feels like it could be based on a helper function or prism for parsing Bech32 encoded hashes of various lengths. But I'm happy to approve as is.